src/CookieTracker.cs

```diff
@@ -20,6 +20,7 @@ public sealed class CookieTracker : IAsyncDisposable
         private readonly Dictionary<Ulid, CachedCookie> _cachedCookies = new();
         private readonly NpgsqlConnection _databaseConnection;
         private readonly ILogger<CookieTracker> _logger;
+        private readonly SemaphoreSlim _dbConnectionSemaphore = new(1, 1);
```

Comment:
Why are we creating a second semaphore slim? If we don't use the original one, then operations will still attempt to execute when the connection could still be broken

Reply:
see [this conversation](https://discord.com/channels/1070516376046944286/1103119299188641802/1104774964059570247) on Discord

Reply:
Here's a snippet of that convo
![Discord_NrGwgurShv](https://user-images.githubusercontent.com/13679909/236689283-86515a4f-207f-44d7-8b2e-939ccf05ff91.png)

Reply:
Ah, I see. I personally think the ideal solution would be to have a method that's called before running any operations, double checking if the database connection is open before returning. Regardless, I can do that another day as my priorities lay within the command handler right now.